edge/pkg/devicetwin/dtmanager/twin_test.go

```diff
@@ -0,0 +1,1534 @@
+			context: &dtcontext.DTContext{},
+			msg:     msgTypeErrorFunc(dttype.BaseMessage{EventID: "Event1"}),
+			errWant: errors.New("msg not Message type"),
+		},
+		{
+			name:    "TestDealTwinGet(): Case 2: invalid message content",
+			context: &dtcontext.DTContext{},
+			msg:     msgTypeFunc(dttype.BaseMessage{EventID: "Event1"}),
+			errWant: errors.New("invalid message content"),
+		},
+		{
+			name:     "TestDealTwinGet(): Case 3: Success case",
+			context:  &context,
+			resource: "DeviceB",
+			msg:      msgTypeFunc(contentKeyTwin),
+			errWant:  nil,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			if _, err := dealTwinGet(test.context, test.resource, test.msg); !reflect.DeepEqual(err, test.errWant) {
+				t.Errorf("DTManager.TestDealTwinGet() case failed: got = %v, Want = %v", err, test.errWant)
+			}
+		})
+	}
+}
+
+// TestDealTwinUpdate is function to test dealTwinUpdate
+func TestDealTwinUpdate(t *testing.T) {
+	content, _ := json.Marshal(dttype.DeviceTwinUpdate{BaseMessage: dttype.BaseMessage{EventID: "Event1"}})
+	context := contextFunc()
+
+	tests := []struct {
+		name     string
+		context  *dtcontext.DTContext
+		resource string
+		msg      interface{}
+		errWant  error
+	}{
+		{
+			name:    "TestDealTwinUpdate(): Case 1: msg not Message type",
+			context: &dtcontext.DTContext{},
+			msg:     msgTypeErrorFunc(dttype.BaseMessage{EventID: "Event1"}),
+			errWant: errors.New("msg not Message type"),
+		},
+		{
+			name:    "TestDealTwinUpdate(): Case 2: invalid message content",
+			context: &dtcontext.DTContext{},
+			msg:     msgTypeFunc(dttype.BaseMessage{EventID: "Event1"}),
+			errWant: errors.New("invalid message content"),
+		},
+		{
+			name:     "TestDealTwinUpdate(): Case 3: Success case",
+			context:  &context,
+			resource: "DeviceB",
+			msg:      msgTypeFunc(content),
+			errWant:  nil,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			if _, err := dealTwinUpdate(test.context, test.resource, test.msg); !reflect.DeepEqual(err, test.errWant) {
+				t.Errorf("DTManager.TestDealTwinUpdate() case failed: got = %v, Want = %v", err, test.errWant)
+			}
+		})
+	}
+}
+
+// TestUpdated is function to test Updated
+func TestUpdated(t *testing.T) {
+	content, _ := json.Marshal(dttype.DeviceTwinUpdate{BaseMessage: dttype.BaseMessage{EventID: "Event1"}})
+	contentKeyTwin, _ := json.Marshal(keyTwinUpdateFunc())
+	context := contextFunc()
+
+	tests := []struct {
+		name     string
+		context  *dtcontext.DTContext
+		deviceID string
+		payload  []byte
+	}{
+		{
+			name:    "TestUpdated(): Case1: Unmarshal update request body failed",
+			context: &dtcontext.DTContext{},
+			payload: content,
+		},
+		{
+			name:    "TestUpdated(): Case2: Success case",
+			context: &context,
+			payload: contentKeyTwin,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			Updated(test.context, test.deviceID, test.payload)
```

Comment:
Is it possible to check whether the test cases are behaving as expected or not after making a call to Updated() ?